roles/network_plugin/cilium/templates/cilium/config.yml.j2

```diff
@@ -232,6 +232,37 @@ data:
   ipam: "{{ cilium_ipam_mode }}"
 {% endif %}
 
+{% if cilium_agent_health_port != "9879" and cilium_version | regex_replace('v') is version('1.11.6', '>=') %}
+  agent-health-port: "{{ cilium_agent_health_port }}"
```

Comment:
You've already added an if clause in the variables. I understand this is meant to add this line only if the default value has changed.

While I haven't tested it out for any possible problems, I feel it should be okay to use ` agent-health-port: "{{ cilium_agent_health_port }}"` and skip the extra lines.